src/function.rs

```diff
@@ -59,6 +60,14 @@ impl Function {
   }
 }
 
+fn abs_path(context: &FunctionContext, path: &str) -> Result<String, String> {
+  let corrected_path = context.invocation_directory.join(path);
+  match corrected_path.canonicalize() {
+    Ok(p) => Ok(p.to_string_lossy().into()),
```

Comment:
Calling to_string_lossy will produce an unusable path, if it isn't valid unicode. I would first convert search.working_directory to unicode:

```
  search.working_directory.to_str().ok_or_else(|| {
    format!(
      "Working directory path is not valid unicode: {}",
      search.working_directory.to_string_lossy()
    )
  })
```

And then turn it into a Utf8Path and join it with `path`

Reply:
I've refactored a little to include this - let me know if it's appropriate.